src/lava/magma/core/model/py/ports.py

```diff
@@ -48,7 +48,18 @@ def peek(self):
         pass
 
     def probe(self) -> bool:
-        pass
+        """Executes probe method of all csp ports and accumulates the returned
+        bool values with AND operation. The accumulator acc is initialized to
+        True.
+
+        Returns the accumulated bool value.
+        """
+        # Returns True only when probe returns True for all _csp_recv_ports.
+        return ft.reduce(
+            lambda acc, csp_port: acc and csp_port.probe(),
+            self._csp_recv_ports,
```

Comment:
Maybe use the property csp_ports instead.

Reply:
I was actually going to, but then saw different other methods in the same file using _csp_recv_ports and _csp_send_ports instead of csp_ports property.
Left _csp_recv_ports usage for homogeneity. Should I change it ?